pkg/deploy/setup.go

```diff
@@ -23,62 +20,40 @@ const (
 
 type Platform string
 
-// isSelfManaged checks presence of ClusterServiceVersions:
-// when CSV displayname contains OpenDataHub, return 'OpenDataHub,nil' => high priority
-// when CSV displayname contains SelfManagedRhods, return 'SelfManagedRhods,nil'
-// when in dev mode and  could not find CSV (deploy by olm), return "", nil
-// otherwise return "",err.
+// isSelfManaged detects if it is Self Managed Rhods or OpenDataHub.
 func isSelfManaged(cli client.Client) (Platform, error) {
-	clusterCsvs := &ofapi.ClusterServiceVersionList{}
-	err := cli.List(context.TODO(), clusterCsvs)
-	if err != nil {
-		return "", err
-	} else { //nolint:golint,revive // Readability on else
-		for _, csv := range clusterCsvs.Items {
-			if strings.Contains(csv.Spec.DisplayName, string(OpenDataHub)) {
-				return OpenDataHub, nil
-			}
-			if strings.Contains(csv.Spec.DisplayName, string(SelfManagedRhods)) {
-				return SelfManagedRhods, nil
-			}
+	variants := map[string]Platform{
+		"opendatahub-operator": OpenDataHub,
+		"rhods-operator":       SelfManagedRhods,
+	}
+
+	for k, v := range variants {
+		exists, err := OperatorExists(cli, k)
+		if err != nil {
+			return Unknown, err
+		}
+		if exists {
+			return v, nil
 		}
 	}
+
 	return Unknown, nil
 }
 
-// isManagedRHODS checks if CRD add-on exists and contains string ManagedRhods.
-func isManagedRHODS(cli client.Client) (Platform, error) {
-	catalogSourceCRD := &apiextv1.CustomResourceDefinition{}
+// isManagedRHODS checks if catsrc CR add-on exists ManagedRhods.
+func isManagedRHODS(ctx context.Context, cli client.Client) (Platform, error) {
+	catalogSource := &ofapi.CatalogSource{}
 
-	err := cli.Get(context.TODO(), client.ObjectKey{Name: "catalogsources.operators.coreos.com"}, catalogSourceCRD)
+	err := cli.Get(ctx, client.ObjectKey{Name: "addon-managed-odh-catalog", Namespace: "openshift-marketplace"}, catalogSource)
```

Comment:
cannot we use some global variables to set the namespace and catalog source name everywhere?